lib/hsa/hsaRfiUpdateBadInputs.cpp

```diff
@@ -51,12 +59,36 @@ hsaRfiUpdateBadInputs::~hsaRfiUpdateBadInputs() {
 int hsaRfiUpdateBadInputs::wait_on_precondition(int gpu_frame_id) {
     (void)gpu_frame_id;
 
-    uint8_t* frame =
-        wait_for_full_frame(_network_buf, unique_name.c_str(), _network_buf_precondition_id);
-    if (frame == nullptr)
-        return -1;
-
-    _network_buf_precondition_id = (_network_buf_precondition_id + 1) % _network_buf->num_frames;
+    // Check for bad input updates
+    if (first_pass) {
+        uint8_t* frame =
+            wait_for_full_frame(_in_buf, unique_name.c_str(), _in_buf_precondition_id);
+        DEBUG("hsaRfiUpdateBadInputs::wait_on_precondition: Bad inputs update.");
+        if (frame == NULL)
+            return -1;
+        _in_buf_precondition_id = (_in_buf_precondition_id + 1) % _in_buf->num_frames;
+        first_pass = false;
+        frame_to_fill = device.get_gpu_buffer_depth();
+        frame_to_fill_finalize = frame_to_fill;
+        filling_frame = true;
+    } else {
+        // Check for new bad inputs only if filled all gpu frames (not currently filling frame)
+        if (!filling_frame) {
+            auto timeout = double_to_ts(0);
+            int status = wait_for_full_frame_timeout(_in_buf, unique_name.c_str(),
+                                                     _in_buf_precondition_id, timeout);
+            DEBUG("status of bad inputs _in_buf_precondition_id[{:d}]={:d} (0=ready 1=not)",
+                  _in_buf_precondition_id, status);
+            if (status == 0) {
+                filling_frame = true;
```

Comment:
There is a mix of up variables here.  These look like they are copied from the `AsyncGain `logic, but haven't been updated in the rest of the code to match the logic.   However given the bug we found in #603, this change should likely merge that in as well, and then the variables here should be updated to reflect the way that code now works.